ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/DiffBasedWorldStateProvider.java

```diff
@@ -141,7 +143,7 @@ public Optional<MutableWorldState> getMutable(
       return cachedWorldStorageManager
           .getWorldState(blockHeader.getHash())
           .or(() -> cachedWorldStorageManager.getNearestWorldState(blockHeader))
-          .or(() -> cachedWorldStorageManager.getHeadWorldState(blockchain::getBlockHeader))
+          .or(() -> cachedWorldStorageManager.getWorldState(chainHeadBlockHeader.getHash()))
```

Comment:
why removing 'head' ?

Reply:
I found a race condition where `getHeadWorldState` can return `empty` because `blockchain.getBlockHeader()` returns empty for the world state block hash in `rootWorldStateStorage`. For archive it's better that we get the latest block that `blockchain` can provide, even if that's not the absolute latest world state. If you think it's better for `DiffBasedWorldStateProvider` to return `Optional.empty` rather than that, I'm happy to revert the change in `DiffBasedWorldStateProvider`.

Reply:
I understand the need. The problem is that with this modification, if your head is not in the cache, it won't work. So I think it's a bit risky. When exactly do you have the race condition? Who is calling the getMutable at that moment? Added that the head is there as a fallback normally if nothing is in the cache. We take the persisted state.